src/adapters/tufts/adapter.js

```diff
@@ -45,16 +46,22 @@ class AlpheiosTuftsAdapter extends BaseAdapter {
    *      - {undefined} - if failed
   */
   async getHomonym (languageID, word) {
-    let url = this.prepareRequestUrl(languageID, word)
-    if (!url) {
-      this.addError(this.l10n.messages['MORPH_TUFTS_NO_ENGINE_FOR_LANGUAGE'].get(languageID.toString()))
-      return
-    }
+    let res
     try {
-      let res = await this.fetch(url)
-      if (res.constructor.name === 'AdapterError') {
-        return
+      if (this.sourceData) {
```

Comment:
Would't that always be true because sourceData is set in the constructor? (but I'm probably missing something here)

Reply:
No, it is not true, from my point of view
```
this.sourceData = config.sourceData
```

So if sourceData is not passed to parameters (it could be from input params or, I don't know if it is useful, but anyway, from default config) it would be `undefined` . And not equal to true.